ldi-core/ldes-client-v2/tree-node-supplier/src/main/java/ldes/client/treenodesupplier/domain/entities/MemberRecord.java

```diff
@@ -0,0 +1,60 @@
+package ldes.client.treenodesupplier.domain.entities;
+
+import ldes.client.treenodesupplier.domain.valueobject.MemberStatus;
+import org.apache.jena.rdf.model.Model;
+
+import java.util.Objects;
+
+public class MemberRecord {
+	private final String memberId;
+
+	private MemberStatus memberStatus;
+	private Model model;
+
+	public MemberRecord(String memberId, Model model) {
+		this.memberId = memberId;
+		this.model = model;
+		this.memberStatus = MemberStatus.UNPROCESSED;
+	}
+
+	public MemberRecord(String memberId, Model model, MemberStatus memberStatus) {
+		this.memberId = memberId;
+		this.model = model;
+		this.memberStatus = memberStatus;
+	}
+
+	public void processedMemberRecord() {
+		this.model = null;
+		this.memberStatus = MemberStatus.PROCESSED;
+	}
+
+	public String getMemberId() {
+		return memberId;
+	}
+
+	public MemberStatus getMemberStatus() {
+		return memberStatus;
+	}
+
+	public SuppliedMember createSuppliedMember() {
+		return new SuppliedMember(memberId, model);
```

Comment:
During creation of SuppliedMember, Jena creates a deep copy.
The reason why SuppliedMember is used, is to avoid that we use MemberRecord for which we erase the model after it has been processed (to save space).

Reply:
Added a test to show the behaviour: `SuppliedMemberTest`